From dd3ba3d596c2b84e2741a6ea7920a444c54c94d3 Mon Sep 17 00:00:00 2001
From: YunYi Yang <yangyunyi2@huawei.com>
Date: Thu, 24 Aug 2023 17:29:53 +0800
Subject: [PATCH 084/256] fix kabi broken due to change of struct blk_mq_tags

driver inclusion
category: bugfix
bugzilla: https://gitee.com/src-openeuler/kernel/issues/I8F7ZR

----------------------------------------------------------------------

Commit ("blk-mq: Use pointers for blk_mq_tags
bitmap tags") changes the struct blk_mq_tags, so we need
to fix kabi broken problem.

Signed-off-by: YunYi Yang <yangyunyi2@huawei.com>
---
 block/blk-mq-tag.h | 14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/block/blk-mq-tag.h b/block/blk-mq-tag.h
index d932f9416d93..ce57fd05a4c2 100644
--- a/block/blk-mq-tag.h
+++ b/block/blk-mq-tag.h
@@ -13,15 +13,21 @@ struct blk_mq_tags {
 
 	atomic_t active_queues;
 
-	struct sbitmap_queue *bitmap_tags;
-	struct sbitmap_queue *breserved_tags;
-
+#ifdef __GENKSYMS__
+	struct sbitmap_queue bitmap_tags;
+	struct sbitmap_queue breserved_tags;
+#else
 	struct sbitmap_queue __bitmap_tags;
 	struct sbitmap_queue __breserved_tags;
-
+#endif
 	struct request **rqs;
 	struct request **static_rqs;
 	struct list_head page_list;
+
+#ifndef __GENKSYMS__
+	struct sbitmap_queue *bitmap_tags;
+	struct sbitmap_queue *breserved_tags;
+#endif
 };
 
 struct blk_mq_tags_wrapper {
-- 
2.27.0

